### DISCUSSION TITLE
[help] `tar_watch()` hangs unexpectedly

### DISCUSSION THREAD

**Discussion by philiporlando** [ropensci/targets#1151] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I've noticed that my local `tar_watch()` session will occasionally freeze during development. I haven't been able to pin point exactly what is causing it to hang. It seems related to running code interactively while debugging. For example, none of the `tar_*()` functions appear to cause issues, but if I try running code line-by-line within one of my function definitions while testing, then my local shiny app will freeze (the screen goes gray, and the refresh bubbles in the top left are stuck in an infinite loop). 

Hoping someone can help me understand what is causing this to happen so I can avoid having to re-run `tar_watch()` a bunch of times! Thank you.

**Comment by wlandau** [comment]
Would you post a reprex and walk me through how to cause the hanging?

**Reply by philiporlando** [reply]
Still working on a reprex. The issue doesn't seem to happen within the small test project that I'm using to build a reprex. My main project is more complicated, and interacts with external resources like databases and shared network drives. I'm wonder if that would have anything to do with `tar_watch()` hanging...

**Reply by philiporlando** [reply]
Ok, I'm noticing that the `tar_watch()` session will hang whenever I call a function that is included within one of my organization's internal R packages. This function includes a call to some Python functions via `reticulate`. I'll try exploring this in more detail to see if a reprex around `reticulate` will help explain the issue further.

**Comment by wlandau** [comment]
Closing because this is an old thread, but let me know if you need anything else.

**Reply by philiporlando** [reply]
Understood. I still haven't figured out how to reliably reproduce the issue. I'll report back if I learn anything new.

**Comment by philiporlando** [comment]
I'm noticing that this issue doesn't happen when calling `tar_watch()` from a separate R process. I still haven't been able to diagnose what is causing it to hang when calling from RStudio.

**Comment by philiporlando** [comment]
I still don't have any answers as to why my `tar_watch()` session would intermittently hang, but my current approach seems to be a nice workaround and I wanted to document it here. 

Instead of calling `tar_watch()` from the RStudio console, I'm creating a dedicated terminal and then launching a separate R session within this terminal that calls `tar_watch()`. 

I've also added the below snippet to my project-local `.Rprofile` so the app always launches with my desired config on a dedicated port:

```r
if (interactive()) {
  tar_watch <- function(...) {
    targets::tar_watch(
      ...,
      display = "graph",
      targets_only = TRUE,
      seconds = 20,
      port = 54716,
      background = FALSE
      )
  }
}
```

I've tested this session over several days and it has been working great for me.
